pkg/deps/config/config.go

```diff
@@ -71,3 +74,55 @@ func NewSupplyBlockClientFn(pocketNodeWebsocketUrl string) SupplierFn {
 		return depinject.Configs(deps, depinject.Supply(blockClient)), nil
 	}
 }
+
+// NewSupplyQueryClientContextFn returns a function with constructs a ClientContext
+// instance with the given cmd and returns a new depinject.Config which is
+// supplied with the given deps and the new ClientContext.
+func NewSupplyQueryClientContextFn(pocketQueryNodeUrl string) SupplierFn {
+	return func(_ context.Context,
+		deps depinject.Config,
+		cmd *cobra.Command,
+	) (depinject.Config, error) {
+		// Set --node flag to the --pocket-node for the client context
+		// This flag is read by cosmosclient.GetClientQueryContext.
+		err := cmd.Flags().Set(cosmosflags.FlagNode, pocketQueryNodeUrl)
+		if err != nil {
+			return nil, err
+		}
+
+		// NB: Currently, the implementations of GetClientTxContext() and
+		// GetClientQueryContext() are identical, allowing for their interchangeable
+		// use in both querying and transaction operations. However, in order to support
+		// independent configuration of client contexts for distinct querying and
+		// transacting purposes. E.g.: transactions are dispatched to the sequencer
+		// while queries are handled by a trusted full-node.
```

Comment:
Tbh, I think we might end up using the pocket network as a full node for doing this. 

It's going to get meta